### PR TITLE
Updated third-party dependencies

### PR DESCRIPTION
**Pull request by JoeHegarty**
This is a general update of third-party dependencies.

### CODE REVIEW

File: actors/runtime/pom.xml [hunk 1]
```diff
@@ -122,12 +122,12 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
         <dependency>
             <groupId>com.esotericsoftware</groupId>
             <artifactId>kryo</artifactId>
-            <version>3.0.3</version>
+            <version>4.0.0</version>
```
Comment (ShawnTuatara) [inline actors/runtime/pom.xml]: Concerns with breaking changes around serialization: https://github.com/EsotericSoftware/kryo/releases

**Comment by ShawnTuatara** [comment]
Would be nice to have context on the need to increase the version for each of the dependencies that are being updated.

File: actors/runtime/pom.xml (hunk 1 above)
Comment (JoeHegarty) [inline actors/runtime/pom.xml]: Yes, I spotted that, but the only data that potentially lives between a deployment is actor directory and we never guarantee compatibility.

**Comment by JoeHegarty** [comment]
There is no specific context beyond keeping up to date with our major dependencies so we don't lag behind. This is to ensure we're getting any security fixes and general improvements and to ensure that there is no expensive integration down the road.

**Comment by JoeHegarty** [comment]
The reason that triggered me to go do this now is the new ea-async which includes support for DCEVM agent loading.

**Pull request merged**